apps/docs/src/components/Search/Markprompt.tsx

```diff
@@ -0,0 +1,126 @@
+import { useEffect, useRef, useState } from 'preact/hooks'
+import hljs from 'highlight.js'
+//@ts-ignore
+import hljs_svelte from 'highlightjs-svelte'
+
+function Search({ visible }: { visible: boolean }) {
+  const container = useRef(null)
+
+  const callback = (mutationList: any) => {
+    for (const mutation of mutationList) {
+      if (
+        mutation.type === 'attributes' &&
+        mutation.attributeName === 'data-loading-state' &&
+        mutation.target.className === 'MarkpromptContentPlain'
+      ) {
+        const value = mutation.target.attributes.getNamedItem('data-loading-state')
+
+        if (value.nodeValue === 'done') {
+          mutation.target.querySelectorAll('pre').forEach((el: HTMLElement) => {
+            if (!el.classList.contains('hljs') && el.innerHTML.length > 20) {
+              hljs.highlightElement(el)
+            }
+          })
+        }
+      }
+    }
+  }
+
+  const config = { attributes: true, childList: true, subtree: true }
+
+  useEffect(() => {
+    const observer = new MutationObserver(callback)
+
+    if (container.current) {
+      observer.observe(container.current, config)
+    }
+
+    return () => observer.disconnect()
+  }, [container.current])
+
+  return (
+    visible && (
+      <>
+        <div
+          className="h-[75vh]"
+          ref={container}
+        >
+          <Markprompt
+            projectKey="q5trazVG0x8ktiY63vB3imwnZJ9bHbyN"
+            close={{
+              visible: false
+            }}
+            search={{
+              enabled: true,
+              provider: {
+                name: 'algolia',
+                apiKey: 'b14257a2e455fc213a51ce63942dd721',
+                appId: 'ZEJ3L4CHC0',
+                indexName: 'threlte'
+              }
+            }}
+            references={{
+              heading: 'Visit sources this answer was generated from:',
+              getHref: (ref) => {
+                const baseUrl = `/docs${ref.file.path.replace('.mdx', '').replace('.md', '')}`
+                if (ref.meta?.leadHeading?.slug) {
+                  return `${baseUrl}#${ref.meta.leadHeading.slug}`
+                }
+                return baseUrl
+              },
+              getLabel: (ref) => {
+                if (JSON.stringify(ref.file.meta.title) === '{}') {
+                  return ref.file.meta.name
+                }
+
+                if (ref.meta?.leadHeading?.value) {
+                  return `${ref.file.meta.category} - ${ref.meta.leadHeading.value}`
+                }
+                return `${ref.file.meta.category} - ${ref.file.meta.title}`
+              }
+            }}
+            display="plain"
+          />
+        </div>
+      </>
+    )
+  )
+}
+
+export default function SearchDialog() {
+  // Visibility
+  const [visible, setVisible] = useState(false)
+
+  const unsubscribe = searchActive.subscribe((v) => {
+    setVisible(v)
+  })
+  useEffect(() => {
+    return () => unsubscribe()
+  })
```

Comment:
Every time the component renders we're declaring a variable "unsubscribe" whereas it should only be declared once after the first render of the component, so it has to be moved into the effect as well, can further be shortened:

```ts
useEffect(searchActive.subscribe(setVisible))
```